### DISCUSSION TITLE
Webhooks for system collections

### DISCUSSION THREAD

**Discussion by scouech** [directus/directus#4835] [Archive]
Is it possible to trigger a webhook when the `directus_users` table gets updated or a new user is added to it? 
I thought when creating a new webhook I could change the collection for `directus_users` but it doesn't seem to work.

Cheers!

**Comment by rijkvanzanten** [comment]
Showing the system collections under that UI makes sense to me! Ideally it'd have the system ones under a sort of "show more" in that interface.

For the time being, you can set the collection directly in the database and it should still work as expected 👍🏻

**Reply by scouech** [reply]
Hey @rijkvanzanten! Coming back to you related to the directus_users webhook. I tried to set it up with the database (see screenshot) but it doesn't work. 
Any idea? 

Cheers
<img width="1205" alt="Screenshot 2021-04-21 at 1 13 53 PM" src="https://user-images.githubusercontent.com/26603052/115544980-90f0c800-a2a3-11eb-8350-4baf828c93d2.png">

**Reply by rijkvanzanten** [reply]
Ohh you're correct.. apologies. The web hooks are registered on the `items.*` hook with a filter to check for included collections. Users are fired on the `users` hook 😭

**Reply by scouech** [reply]
so it's not possible at the moment? 🤓

**Reply by rijkvanzanten** [reply]
No I'm afraid not. That being said, a webhook is nothing more than a regular hook that makes a HTTP request, so it should be fairly straightforward to setup through a hook extension 👍🏻

**Reply by scouech** [reply]
thanks @rijkvanzanten ! I don't think I have enough knowledge to build a hook.
Do you think it would be possible to add the feature itself to directus in the future? Basically not restrict webhooks to items only?

**Reply by rijkvanzanten** [reply]
Yeah! We'll have to update this bit:

https://github.com/directus/directus/blob/f836c90990859781cf9af32bdf8ae40fbd7d75b5/api/src/webhooks.ts#L15-L30

to figure out the `items` part dynamically, instead of being hardcoded to `items.*`.

**Reply by scouech** [reply]
mmm if you want I could look into it in the next 2 weeks or something! :)
